cloud/amazon/ec2_vpc_route_table.py

```diff
@@ -320,17 +320,17 @@ def ensure_routes(vpc_conn, route_table, route_specs, propagating_vgw_ids,
                         if r.gateway_id != 'local'
                         and r.gateway_id not in propagating_vgw_ids]
 
-    changed = routes_to_delete or route_specs_to_create
+    changed = bool(routes_to_delete or route_specs_to_create)
     if changed:
-        for route_spec in route_specs_to_create:
-            vpc_conn.create_route(route_table.id,
-                                  dry_run=check_mode,
-                                  **route_spec)
-
         for route in routes_to_delete:
             vpc_conn.delete_route(route_table.id,
                                   route.destination_cidr_block,
                                   dry_run=check_mode)
+
+        for route_spec in route_specs_to_create:
+            vpc_conn.create_route(route_table.id,
+                                  dry_run=check_mode,
+                                  **route_spec)
```

Comment:
Would `create_route` previously fail if there were conflicts, and what would a conflict look like? Or would it update them? This new change can potentially delete routes and fail to create their replacement, if creating any other preceding route fails. That's a potentially a bad failure mode.